docs/articles/getting-started/batch-example.md

```diff
@@ -1,68 +1,171 @@
-# Example batch operator
-
-Showing the use of batch and simple window
-
-## Stateful Operators and Keyed Streams
-
-Bytewax has **stateful operators** that allow you to persist state
-between items flowing in the dataflow. Some examples are
-`bytewax.operators.window.fold_window` and
-`bytewax.operators.stateful_map`. The Bytewax runtime partitions state
-across multiple workers by a **state key**. This means that for a
-given key, all items are moved onto the primary worker which has the
-state for the key before processing the step. This means the state
-overall is balanced over all workers, allowing rescaling, but the
-state for a single key is localized.
-
-This means that all stateful operators only work with a **keyed
-stream** upstream with items that are shaped like 2-tuples of `(key,
-value)`. Whenever we refer to a **value** in reference to stateful
-operators, it means specifically the second value of these tuples. The
-operators `bytewax.operators.key_on` and
-`bytewax.operators.key_assert` help you make streams of this
-`bytewax.dataflow.KeyedStream` shape. Downstreams of the stateful
-operators will also be keyed and the relevant key will automatically
-be applied.
-
-For example, let's write a small dataflow that calculates a running
-sum of points in a basketball game. First, let's define a
+# Batching and Windowing
+
+In this section, we'll be talking about the Batching and Windowing operators
+that Bytewax provides.
+
+In this context, **Batching** refers to an operation over an unbounded stream
+of data, rather than an execution mode where the amount of data is finite.
+
+It is important to remember that all **Batching** and **Windowing** operators
+collect data based on a key and therefore expect input `Streams` to be keyed.
+
+
+## Batching
+
+**Batching** operators operate over a stream of data and collect a number of
+items until a fixed size, or a given timeout is reached.
+
+Let's construct a simple dataflow to demonstrate. In the following Dataflow,
+we're using the `TestingSource` **Source** to generate data from a list,
+but in a production Dataflow, your input could come from RedPanda, or another
+other streaming input source.
+
+The `TestingSource` emits one integer at a time into the dataflow. In our
+batch operator, we'll configure it to wait for either three values, or
+for 10 seconds to expire before emitting the batch downstream. Since
+we won't be waiting for data with our `TestingSource`, we should
+always see batches of 3 items.
 
 ```python
-from bytewax.dataflow import Dataflow
+from datetime import timedelta
+
 import bytewax.operators as op
+import bytewax.operators.window as win
+from bytewax.dataflow import Dataflow
+from bytewax.connectors.stdio import StdOutSink
 from bytewax.testing import TestingSource
-import bytewax.testing
+
+flow = Dataflow("batching")
+s = op.input("input", flow, TestingSource(list(range(10))))
+# The following line creates a keyed stream where all values have the key "ALL"
```

Comment:
I'd comment on why we are doing this rather than what we are doing, something like:
```python
# Here we want to batch all the items together, so we use the same fixed key for all the items
```